br/pkg/encryption/master_key/file_backend.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2024 PingCAP, Inc. Licensed under Apache-2.0.
+
+package encryption
+
+import (
+	"context"
+	"encoding/hex"
+	"os"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/kvproto/pkg/encryptionpb"
+)
+
+const AesGcmKeyLen = 32 // AES-256 key length
+
+type FileBackend struct {
+	memCache *MemAesGcmBackend
+}
+
+func createFileBackend(keyPath string) (*FileBackend, error) {
+	// FileBackend uses AES-256-GCM
+	keyLen := AesGcmKeyLen
+
+	content, err := os.ReadFile(keyPath)
+	if err != nil {
+		return nil, errors.Annotate(err, "failed to read master key file from disk")
+	}
+
+	fileLen := len(content)
+	expectedLen := keyLen*2 + 1 // hex-encoded key + newline
```

Comment:
yes this is proved from TiKV

```
impl FileBackend {
    pub fn new(key_path: &Path) -> Result<FileBackend> {
        // FileBackend uses Aes256-GCM.
        let key_len = AesGcmCrypter::KEY_LEN;
        let mut file = File::open(key_path)?;
        // Check file size to avoid reading a gigantic file accidentally.
        let file_len = file.metadata()?.len() as usize;
        if file_len != key_len * 2 + 1 {
            return Err(box_err!(
                "mismatch master key file size, expected {}, actual {}.",
                key_len * 2 + 1,
                file_len
            ));
        }
```